src/main/java/org/datadog/jenkins/plugins/datadog/clients/DogStatsDClient.java

```diff
@@ -0,0 +1,210 @@
+package org.datadog.jenkins.plugins.datadog.clients;
+
+import com.timgroup.statsd.*;
+import hudson.util.Secret;
+import net.sf.json.JSONObject;
+import org.datadog.jenkins.plugins.datadog.DatadogClient;
+import org.datadog.jenkins.plugins.datadog.DatadogEvent;
+import org.datadog.jenkins.plugins.datadog.util.TagsUtil;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * This class is used to collect all methods that has to do with transmitting
+ * data to Datadog.
+ */
+public class DogStatsDClient implements DatadogClient {
+
+    private static DatadogClient instance;
+    private static final Logger logger = Logger.getLogger(DatadogHttpClient.class.getName());
+
+    public static boolean enableValidations = true;
+
+    private StatsDClient statsd;
+    private String hostname;
+    private int port = -1;
+    private boolean isStopped = true;
+
+    /**
+     * NOTE: Use DatadogUtilities.getDatadogClient method to instantiate the client in the Jenkins Plugin
+     * This method is not recommended to be used because it misses some validations.
+     * @param url - target url
+     * @param port - target port
+     * @return an singleton instance of the DatadogClient.
+     */
+    public static DatadogClient getInstance(String url, Integer port){
+        if(enableValidations){
+            if (url == null || url.isEmpty()) {
+                logger.severe("Datadog Target URL is not set properly");
+                throw new RuntimeException("Datadog Target URL is not set properly");
+            }
+        }
+
+        if(instance == null){
+            synchronized (DatadogHttpClient.class) {
+                if(instance == null){
+                    instance = new DogStatsDClient( url, port);
+                }
+            }
+        }
+
+        // We reset param just in case we change values
+        instance.setApiKey(null);
+        instance.setUrl(url);
```

Comment:
I don't understand this

Reply:
You don't understand because it is indeed wrong 😆